It is assumed that the mean systolic blood pressure is μ = 120 mm Hg. In the Honolulu Heart Study, a sample of n = 100 people had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. Is the group significantly different (with respect to systolic blood pressure!) from the regular population?


#### testing whether the population mean fits inside the confidence interval of my sample 

### Set up the hypothesis test


In [1]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', None)
import warnings
warnings.filterwarnings('ignore')
import matplotlib.pyplot as plt
import seaborn as sns 
from scipy import stats

Null-Hypothesis:
    
The mean systolic blood pressure of the Honolulu Heart Study is not signiciantly different from the regular poplation

H0: μ = X

Alternative Hypthesis:
    
The mean systolic blood pressure of the Honolulu Heart Study is signiciantly different from the regular poplation
    
Ha: μ != X


Since μ could be higher or lower than the population mean I use a two tail test

#### (Background Infos for me)

Assuming data is normally distributed and the number of observations are small, that the data is continuous and we are focused on the mean difference, we can use a t-test. This test is based on a "student’s t distribution" which is very similar to standard normal distribution, except it is much flatter on tails.


What’s the key difference between the t- and z-distributions?
The standard normal or z-distribution assumes that you know the population standard deviation. 
The t-distribution (known as the student’s t-distribution) is based on the sample standard deviation.


Level of significance: This defines the rejection region - the probability of rejecting H0 if it were true. It is denoted by alpha or α, conventionally 0.05 or 0.01


In [4]:
# Calculating T statistic for one sample t test

import math

sample_mean = 130.1
pop_mean = 120
sample_std = 21.21
n = 100
statistic = (sample_mean - pop_mean)/(sample_std/math.sqrt(n))
print("Statistic is: ", statistic)

T Statistic is:  4.761904761904759


#### The statistic is the amount of distances between the two values I am testing (Here pop mean and sample mean)
the distance is measures as std divided by the square root of n

A critical value is said to be as a line on a graph that divides a distribution graph into sections that indicate ‘rejection regions.’


Degrees of freedom of an estimate is the number of independent pieces of information that went into calculating the estimate. It’s not quite the same as the number of items in the sample. In order to get the df for the estimate, you have to subtract 1 from the number of items


**Degrees of freedom** is 99

**Hence the criticial value is** 

-1.9842 on the left hand side of the value that I want to test, in this case the mean

+1.9842 on the right hand side of the value that I want to test, in this case the mean


according to https://calculator-online.net/critical-value-calculator/

The distance measured as Statstic( 4.76) is not falling into my confidence interval, which goes from -1.9842 to +1.9842, hence we reject the Null-Hypthesis. the pop mean and the sample mean are significantly different. 

### Doing the same with p value

In [7]:
# Using python to find the p value and critical value

from scipy.stats import t

print("P value is: ", 1- t.cdf(statistic,n))
print("Critical Value of z is: ", t.ppf(0.025, n)) #alpha is 0.05

P value is:  3.2462138190858525e-06
Critical Value of z is:  -1.983971518449634
